src/main/java/uk/gov/justice/digital/delius/service/ConvictionService.java

```diff
@@ -114,6 +114,17 @@ public ConvictionService(
                 .collect(toList());
     }
 
+    @Transactional(readOnly = true)
+    public Optional<Conviction> convictionFor(Long offenderId, Long eventId) {
+        List<uk.gov.justice.digital.delius.jpa.standard.entity.Event> events = eventRepository.findByEventId(eventId);
```

Comment:
once you use `findById` you can get rid of the `.stream()` and `.findFirst` below. And very minor prefer `final var` to typed left hand argument.

Reply:
Yes